lib/src/util/component_usage.dart

```diff
@@ -495,29 +495,35 @@ class IndexPropAssignment extends BuilderMemberAccess {
   Expression get rightHandSide => node.rightHandSide;
 }
 
-/// Returns whether [expression] and all of its descendants are fully resolved
-/// by checking whether there are any unresolved identifiers or unresolved
-/// invocations.
+/// Returns whether [expression] or any of its descendants contain references
+/// that either:
 ///
-/// This is a more accurate alternative than just checking for [Expression.staticType],
-/// since that often shows up as `dynamic` in resolved contexts when something doesn't fully resolve.
-bool isFullyResolved(Expression expression) {
-  if (expression.staticType == null) return false;
+/// 1. are not fully resolved
+/// 2. represent members accessed on `dynamic` expressions (e.g., `dynamicVar.foo`)
+///
+/// Unfortunately, it's difficult to distinguish between the two and only detect
+/// unresolved references.
+///
+/// If dynamic member accesses are rare, this can a more accurate alternative
+/// than just checking for [Expression.staticType], since that often shows up
+/// as `dynamic` in resolved contexts when something doesn't fully resolve.
+bool hasUnresolvedOrDynamicReferences(Expression expression) {
```

Comment:
This method and related code were updated to return the boolean inverse of what it used to return due to the name being updated.